libraries/AP_BoardConfig/AP_BoardConfig.cpp

```diff
@@ -97,6 +102,16 @@ const AP_Param::GroupInfo AP_BoardConfig::var_info[] = {
     AP_GROUPINFO("CAN_ENABLE", 6, AP_BoardConfig, _can_enable, 0),
 #endif
     
+#if CONFIG_HAL_BOARD == HAL_BOARD_PX4
```

Comment:
I can separate if if you think that is better. I actually like that explicitness/short scope of this, (I have a dislike for nested #if's in a scenario like this. I'm happy to go with the preferred style though.